### PR TITLE
AF-1799: Upgrade JDK 11 + GWT 2.9

### PR DESCRIPTION
**Pull request by caponetto**
**Upgrade JDK 11 + GWT 2.9**

Feel free to comment and/or raise any concerns.

This PR continues the work done in 
- https://github.com/kiegroup/appformer/pull/1030
- https://github.com/kiegroup/appformer/pull/1039

Thanks guys for that!

**JIRA**:

[AF-1799](https://issues.redhat.com/browse/AF-1799)

**Parent Pull Request**:
- https://github.com/kiegroup/lienzo-core/pull/134

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by caponetto** [comment]
Jenkins run fdb
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by mareknovotny** [comment]
Jenkins do fdb

**Comment by caponetto** [comment]
JAVA11-test

File: pom.xml [hunk 1]
```diff
@@ -320,7 +320,7 @@
     <version.org.jboss.arquillian.extension.drone>2.5.1</version.org.jboss.arquillian.extension.drone>
     <version.org.jboss.arquillian.graphene>2.3.2</version.org.jboss.arquillian.graphene>
     <version.com.google.guava.for-selenium>23.0</version.com.google.guava.for-selenium>
-    <version.org.jboss.errai>4.10.0.Final</version.org.jboss.errai>
+    <version.org.jboss.errai>4.11.0-SNAPSHOT</version.org.jboss.errai>
```
Comment (mareknovotny) [inline pom.xml:323]: @tiagobento the snapshot from ERRAI master is enough or do you need changes from https://github.com/errai/errai/pull/407 ? the CI here will not take the changes from your PR until you merge it and that master head with Snapshot will be deployed by CI. Just to remind it here.

Comment (tiagobento) [reply pom.xml:323]: @mareknovotny Thanks for the reminder, Marek! We'll probably need to release a new version from Errai though. I'm still validating this solution with other people involved on this upgrade, but I'll keep you updated on the PRs!

**Comment by tiagobento** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

File: pom.xml (hunk 1 above)
Comment (caponetto) [reply pom.xml]: Errai version upgraded to 4.11.0.Final. https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1488/commits/267c92ed8a9a7c84ca5baf8378b54fb757730b92

**Comment by mbiarnes** [comment]
@caponetto please don't forget here also to upgrade the errai version : 4.11.0.Final

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Review by mbiarnes — APPROVED** [review]
+1

**Review by mbiarnes — APPROVED** [review]
+1

**Comment by mbiarnes** [comment]
Jenkins execute fdb

File: pom.xml [hunk 2]
```diff
@@ -5343,18 +5351,21 @@
...
       </dependency>
       
-      <!-- Used by jbpm-workitems-rest -->
+      <!-- Used by jgit -->
       <dependency>
         <groupId>org.bouncycastle</groupId>
         <artifactId>bcprov-jdk15on</artifactId>
         <version>${version.org.bouncycastle}</version>
-        <scope>test</scope>
       </dependency>
```
Comment (mareknovotny) [inline pom.xml:5359]: this is a little bit controversial change as it was only test dependency as noted above and beside that excluded from transitive chains of our deps.

Comment (mareknovotny) [reply pom.xml:5359]: do we really need to manage it and do not exclude it from jgit dependency?

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

File: pom.xml (hunk 2 above)
Comment (caponetto) [reply pom.xml]: `test` scope set for `bouncycastle` dependencies in https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1488/commits/e6bdce2971bf3ba12dcb5e5c1486b343b2f52362.

**Comment by caponetto** [comment]
Once again, we got green FDBs for both JDK8 ([137](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/fdb/job/droolsjbpm-build-bootstrap-master.fdb/137/)) and JDK11 ([138](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/fdb/job/droolsjbpm-build-bootstrap-master.fdb/138/)). :)

**Review by mareknovotny — APPROVED** [review]
lgtm

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
jenkins run fdb

**Review by Srihari1192 — APPROVED** [review]

**Comment by caponetto** [comment]
Jenkins run cdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Comment by caponetto** [comment]
Jenkins run fdb

**Pull request merged**